src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -1596,6 +1596,30 @@ int MPL_gpu_query_pointer_attr(const void *ptr, MPL_pointer_attr_t * attr)
     goto fn_exit;
 }
 
+int MPL_gpu_query_pointer_is_dev(const void *ptr, MPL_pointer_attr_t * attr)
+{
+    ze_result_t ret ATTRIBUTE((unused));
+    ze_memory_type_t type;
+
+    if (attr == NULL) {
+        ze_memory_allocation_properties_t prop = {
+            .stype = ZE_STRUCTURE_TYPE_MEMORY_ALLOCATION_PROPERTIES,
+            .pNext = NULL,
+            .type = 0,
+            .id = 0,
+            .pageSize = 0,
+        };
+        ze_device_handle_t device = NULL;
+
+        ret = zeMemGetAllocProperties(ze_context, ptr, &prop, &device);
+        assert(ret == ZE_RESULT_SUCCESS);
+        type = prop.type;
+    } else {
+        type = attr->device_attr.prop.type;
+    }
+    return type != ZE_MEMORY_TYPE_UNKNOWN;
```

Comment:
This comparison in the return statement is confusing. Wouldn't it return true for host memory?
> The application may query attributes of a memory allocation unrelated to the context. When this occurs, the returned allocation type will be [ZE_MEMORY_TYPE_UNKNOWN](https://spec.oneapi.io/level-zero/latest/core/api.html#ze__api_8h_1a6808370b64ade7fdd4a74093224e1d9ba437d57ae7d0652068b62eabdc6cda93a), and the returned identifier and associated device is unspecified.

Reply:
Oh man. OK, can we stick a comment here basically saying that?